docs/fastddsgen/usage/usage.rst

```diff
@@ -81,28 +75,41 @@ Where the option choices are:
        This option expects the location of the template and the location of the file where source code output will be
        stored. |br|
        A custom template example can be found in this `link <https://raw.githubusercontent.com/eProsima/Fast-DDS-Gen/master/resources/Custom.stg>`_
+   * - -flat-output-dir
+     - Ignore input files relative paths and place all generated files in the specified output directory.
+   * - -fusion
+     - Activates fusion.
    * - -help
      - Shows the help information
    * - -I <directory>
      - Add directory to preprocessor include paths.
+   * - -language <lang>
+     - chooses between <c++> or <java> languages.
+   * - -no-typesupport
+     - Avoid generating the type support files.
+   * - -no-typeobjectsupport
+     - Avoid generating the TypeObject support specific files. |br|
+       Enabled automatically if -no-typesupport argument is used.
+   * - -no-dependencies
+     - Avoid processing the dependent IDL files.
+   * - -package
+     - Default package used in Java files.
    * - -ppDisable
      - Disables the preprocessor.
    * - -ppPath
      - Specifies the preprocessor path.
    * - -python
      - Generates source code and a CMake solution to compile a library containing the data types |br|
        Python bindings required to run a *Fast DDS* Python-based application.
-       This option is |br|
-       incompatible with the `-example` and `-typeobject` ones.
+       This option is incompatible with `-example`.
    * - -replace
      - Replaces the generated source code files even if they exist.
    * - -t <directory>
      - Sets a specific directory as a temporary directory.
-   * - -typeobject
-     - Generates `TypeObject` files for the IDL provided and modifies MyType constructor to |br|
-       register the TypeObject representation into the factory.
    * - -typeros2
-     - Generates type naming compatible with ROS 2
+     - Generates type naming compatible with ROS 2.
+   * - -typesc
```

Comment:
We have to talk about this

Reply:
This option should not be documented and I will remove it

Reply:
Done in 599a362c6cc12101fed9ddf09b4eaa5796a96cf0